### PR TITLE
Update liberal EFM disclosure systems so that they allow EFM 6.22.00 errors for all-years

### PR DESCRIPTION
**Pull request by derekgengenbacher-wf**
#### Reason for change
Need more flexibility between currently existing disclosure systems in the EFM validation plugin

#### Description of change
- Updated `efm-liberal` to use up-to-date edgartaxonomies file
- Added `efm-liberal-all-years` which utilizes a new field on the disclosure system config to determine which taxonomies are valida but allow all taxonomies to be validated with.

#### Steps to Test
CI

**review**:
@Arelle/arelle

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: arelle/DisclosureSystem.py [hunk 1]
```diff
@@ -335,6 +341,45 @@ def loadStandardTaxonomiesDict(self):
                                                  level=logging.ERROR)
                 etree.clear_error_log()
 
+    def loadValidTaxonomiesDict(self):
+        if self.selection:
+            self.validTaxonomiesDict = defaultdict(set)
```
Comment (aaroncameron-wk) [inline arelle/DisclosureSystem.py:346]: This is probably pre-existing from the above method, but I think having `validTaxonomiesDict` be a `defaultdict` could cause issues. Some other code might unknowingly add a key to the dictionary just by inspecting it.

File: arelle/plugin/validate/EFM/config.xml [hunk 2]
```diff
@@ -205,6 +203,45 @@
      standardTaxonomyDatabase="efmStandardTaxonomies.db"
      standardTaxonomyUrlPattern="http://www.fasb.org|http://xbrl.sec.gov|http://www.xbrl.org|http://taxonomies.xbrl.us|http://xbrl.us/us-gaap|http://xbrl.ici.org/rr|http://xbrl.ifrs.org/taxonomy"
      />
+  <DisclosureSystem
+     names="US SEC (Edgar Filing Manual, Liberal, All Years)|efm-nonblocking-all-years|efm-liberal-all-years"
```
Comment (austinmatherne-wk) [inline arelle/plugin/validate/EFM/config.xml:207]: The first name (`US SEC (Edgar Filing Manual, Liberal, All Years)`) conflicts with the entry above `efm-liberal`

File: arelle/DisclosureSystem.py (hunk 1 above)
Comment (derekgengenbacher-wf) [reply arelle/DisclosureSystem.py:346]: Agree, would rather try and fix that in a different PR than this however.

**Review by aaroncameron-wk — APPROVED** [review]

File: arelle/plugin/validate/EFM/config.xml [hunk 3]
```diff
@@ -120,8 +120,6 @@
       deiDocumentPeriodEndDateElement="DocumentPeriodEndDate"
       deiFilerIdentifierElement="EntityCentralIndexKey"
       deiFilerNameElement="EntityRegistrantName"
-      logLevelFilter="(?!.*-semantic$)"
-      logCodeFilter="(?!EFM.6.03.03.matchInstance|EFM.6.05.28.linkrole|EFM.6.07.06|EFM.6.10.01.missingLabelLinkbase)"
```
Comment (austinmatherne-wk) [inline arelle/plugin/validate/EFM/config.xml:124]: Why were these removed? I would've expected log/level filters to align with `efm-pragmatic` not `efm-strict`

**Review by austinmatherne-wk — APPROVED** [review]

**Review by allenbross-wf — APPROVED** [review]

File: arelle/DisclosureSystem.py (hunk 1 above)
Comment (hermfischer-wf) [reply arelle/DisclosureSystem.py:346]: I can't find any place, outside of DisclosureSystem.py standardTaxonomiesDict is used as a dict, just a a set with in.  Suggest changing the externally referenced standardTaxonomiesDict into an frozen set (so no user messes with contents).

Comment (hermfischer-wf) [reply arelle/DisclosureSystem.py:346]: Can't there be one method used for both loadStandardTaxonomiesDict and loadValidTaxonomiesDict, passing in the resulting  output dict or something unduplicative?

File: arelle/DisclosureSystem.py [hunk 4]
```diff
@@ -385,3 +430,10 @@ def hrefValid(self, href):
         if self.standardTaxonomiesUrl:
             return href in self.standardTaxonomiesDict
         return True # no standard taxonomies to test
+
+    def hrefValidForValidTaxonomy(self, href):
```
Comment (hermfischer-wf) [inline arelle/DisclosureSystem.py:434]: What about a less duplicative name, like maybe hrefValidForDisclosureSystem?

File: arelle/DisclosureSystem.py [hunk 5]
```diff
@@ -335,6 +341,45 @@ def loadStandardTaxonomiesDict(self):
                                                  level=logging.ERROR)
                 etree.clear_error_log()
 
+    def loadValidTaxonomiesDict(self):
```
Comment (hermfischer-wf) [inline arelle/DisclosureSystem.py:344]: Can't there be one method used for both loadStandardTaxonomiesDict and loadValidTaxonomiesDict, passing in the resulting output dict or something unduplicative?

File: arelle/ModelDocument.py [hunk 6]
```diff
@@ -72,7 +72,7 @@ def load(modelXbrl, uri, base=None, referringElement=None, isEntry=False, isDisc
             modelXbrl.uriDir = os.path.dirname(modelXbrl.uriDir)
     if modelXbrl.modelManager.validateDisclosureSystem and \
        not normalizedUri.startswith(modelXbrl.uriDir) and \
-       not modelXbrl.modelManager.disclosureSystem.hrefValid(normalizedUri):
+       not modelXbrl.modelManager.disclosureSystem.hrefValidForValidTaxonomy(normalizedUri):
         blocked = modelXbrl.modelManager.disclosureSystem.blockDisallowedReferences
```
Comment (hermfischer-wf) [inline arelle/ModelDocument.py:76]: I thought, for non-blocking, there was a desire to provide a warning or info notice if a href'ed taxonomy were not in the currently accepted list of taxonomies, in which case this logic might need some extra TLC?

Comment (derekgengenbacher-wf) [reply arelle/ModelDocument.py:76]: No we want all the other disclosure systems to remain the way they were. We want to add a singular disclosure system for `all-years` that can throw the EFM 6.22.00 errors.

File: arelle/DisclosureSystem.py (hunk 5 above)
Comment (derekgengenbacher-wf) [reply arelle/DisclosureSystem.py:344]: Agree that this is duplicative. Will look to see how we can remove the duplicity in a follow up PR.

**Review by aaroncameron-wk — APPROVED** [review]

**Review by austinmatherne-wk — APPROVED** [review]

**Review by allenbross-wf — APPROVED** [review]
